controller.go

```diff
@@ -168,6 +242,9 @@ func (c *controller) NewNetwork(networkType, name string, options ...NetworkOpti
 	}
 
 	network.processOptions(options...)
+	if err := c.addNetworkToStore(network); err != nil {
+		log.Debugf("Failed to add network into datastore %v", err)
```

Comment:
We cannot just log an error message to syslog and proceed when `addNetworkToStore` fails. We need to fail the call which means you need to handle the data store not present case in `addNetworkToStore` itself.